gensim/models/sent2vec.py

```diff
@@ -0,0 +1,132 @@
+                checked_string_types += 1
+            if sentence_no % progress_per == 0:
+                logger.info(
+                    "PROGRESS: at sentence #%i, processed %i words, keeping %i word types",
+                    sentence_no, sum(itervalues(vocab)) + total_words, len(vocab)
+                )
+            for word in sentence:
+                num_discarded = 0
+                line_size = len(sentence)
+                discard = [False] * line_size
+                while (num_discarded < self.dropoutK and line_size - num_discarded > 2):
+                    token_to_discard = randint(0,line_size-1)
+                    if discard[token_to_discard] == False:
+                        discard[token_to_discard] = True
+                        num_discarded += 1
+                for i in range(line_size):
+                    if discard[i]:
+                        continue
+                    for j in range(i + self.word_ngrams):
+                        if j >= line_size or discard[j]:
+                            break
+                        vocab[word] += 1
+
+            if self.max_vocab_size and len(vocab) > self.max_vocab_size:
+                total_words += utils.prune_vocab(vocab, min_reduce, trim_rule=trim_rule)
+                min_reduce += 1
+
+        total_words += sum(itervalues(vocab))
+        logger.info(
+            "collected %i word types from a corpus of %i raw words and %i sentences",
+            len(vocab), total_words, sentence_no + 1
+        )
+        self.corpus_count = sentence_no + 1
+        self.raw_vocab = vocab
+
+    def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None, progress_per=10000, update=False):
+        if update:
+            if not len(self.wv.vocab):
+                raise RuntimeError("You cannot do an online vocabulary-update of a model which has no prior vocabulary. "
+                    "First build the vocabulary of your model with a corpus "
+                    "before doing an online update.")
+            self.old_vocab_len = len(self.wv.vocab)
+            self.old_hash2index_len = len(self.wv.hash2index)
+
+        self.scan_vocab(sentences, progress_per=progress_per, trim_rule=trim_rule)
+        self.scale_vocab(keep_raw_vocab=keep_raw_vocab, trim_rule=trim_rule, update=update)
+        self.finalize_vocab(update=update)
+        self.init_ngrams(update=update)
+
+    def train(self, sentences, total_examples=None, total_words=None,
+              epochs=None, start_alpha=None, end_alpha=None,
+              word_count=0, queue_factor=2, report_delay=1.0):
+        super(Sent2Vec, self).train(sentences, total_examples=total_examples, epochs=epochs, start_alpha=start_alpha, end_alpha=end_alpha)
+
+    def __getitem__(self, word):
+        return self.word_vec(word)
+
+    def word_vec(self, word, use_norm=False):
+        return FastTextKeyedVectors.word_vec(self.wv, word, use_norm=use_norm)
+
+    def sent_vec(self, sentence):
```

Comment:
This method is bad, don't need to split the raw string to tokens, should pass for example list of tokens.